### PR TITLE
Misc fixes and improvements

### PR DESCRIPTION
**Pull request by jacmet**
Peter Korsgaard (3):
      httpclient: seperate response headers from body
      httpclient: perform chunked decoding inline
      httpclient: pass body size to callback for binary data handling

### CODE REVIEW

File: httpclient.c [hunk 1]
```diff
@@ -317,21 +314,22 @@ static void ICACHE_FLASH_ATTR disconnect_callback(void * arg)
 			}
 			else {
 				http_status = atoi(req->buffer + strlen(version));
-				body = (char *)os_strstr(req->buffer, "\r\n\r\n") + 4;
+				/* find body and zero terminate headers */
+				body = (char *)os_strstr(req->buffer, "\r\n\r\n") + 2;
+				*body++ = '\0';
```
Comment (Caerbannog) [inline httpclient.c:319]: I like this trick!

File: httpclient.c [hunk 2]
```diff
@@ -496,11 +494,11 @@ void ICACHE_FLASH_ATTR http_get(const char * url, const char * headers, http_cal
 	http_post(url, NULL, headers, user_callback);
 }
 
-void ICACHE_FLASH_ATTR http_callback_example(char * response, int http_status, char * full_response)
+void ICACHE_FLASH_ATTR http_callback_example(char * response, int http_status, char * response_headers, int size)
```
Comment (Caerbannog) [inline httpclient.c:497]: rename last arg to body_size?

File: httpclient.h [hunk 3]
```diff
@@ -45,6 +45,6 @@ void ICACHE_FLASH_ATTR http_raw_request(const char * hostname, int port, bool se
 /*
  * Output on the UART.
  */
-void http_callback_example(char * response, int http_status, char * full_response);
+void http_callback_example(char * response, int http_status, char * response_headers, int response_size);
```
Comment (Caerbannog) [inline httpclient.h:48]: rename last arg to body_size?

**Comment by Caerbannog** [comment]
This commit seems like a good idea, thanks.
Just one additional question:

What are the exact reasons why you added the `body_size` argument to the callback? If the user needs this information, he can strlen() on `response`.

What I am reluctant to do is change the prototype of the API if we don't need it, this will probably break people's code.

**Comment by jacmet** [comment]
As I explained in the commit message, the reason is for binary data (E.G. I've used httpclient to download bitmaps and display on a small OLED display) which may contain embedded zeros, so you cannot just do strlen().

File: httpclient.c (hunk 2 above)
Comment (jacmet) [inline httpclient.c:497]: Ok, I can do that. Do you also want me to rename response to body?

Comment (Caerbannog) [inline httpclient.c:497]: Oh, yeah

**Pull request merged**